### ISSUE TITLE
Update Stripe API libraries

### ISSUE THREAD

**Issue by pippinsplugins** [restrictcontentpro/restrict-content-pro#522]
It's time to get them up to date.

**Comment by mindctrl** [comment]
This looks good to me.

**Comment by michaelbeil** [comment]
All good hear.

**Comment by tnorthcutt** [comment]
Reopening this, as the current Stripe library version is 6.1.0 as of this writing, but RCP includes version 3.2.3, which was released 17 months ago.

I discovered this because I'm currently experiencing [this bug](https://github.com/stripe/stripe-php/issues/289), which was quite a fun experience to track down 😁 

I'm not sure what all implications there might be of updating the Stripe library, nor what test coverage there is that would catch any problems.

I'll prepare a PR shortly regardless, so we have something to work from.
